crypto/src/main/java/org/hyperledger/besu/crypto/PublicKey.java

```diff
@@ -0,0 +1,121 @@
+package org.hyperledger.besu.crypto;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.MutableBytes;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.math.ec.ECPoint;
+import org.bouncycastle.math.ec.FixedPointCombMultiplier;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class PublicKey implements java.security.PublicKey {
```

Comment:
Name should be more descriptive, like `SECPPublicKey`

Reply:
Good suggestion. I've renamed it.